### DISCUSSION TITLE
alias to subcommand

### DISCUSSION THREAD

**Discussion by tahmid-23** [Minestom/Minestom#941] [Help] [answered]
if I have a command with a registered subcommand that works like
`/command subcommand <argument>`,
then how would you be able to make
`/command <argument>` redirect to the subcommand?
executing `/command Hello` for example would be the same as `/command subcommand Hello`
of course I can duplicate the code by having two command classes and executors, but is there a way to do this with the redirect command feature?

**Comment by ArvedEh** [comment, marked as answer]
So lets say your subcommand is called "NoCooldown".

Maybe try this code in the main command:
```java
//  In the main command:
NoCooldown subcommand = new NoCooldown();
setDefaultExecutor((sender, context) -> {
    subcommand.getDefaultExecutor().apply(sender, context);
});
addSubcommand(subcommand);
```

Here are a few things to look out for:
- The subcommand should only use the context to get their arguments, and not to get any other info (like the command name) because those will be inaccurate.
- You will have to register this syntax just like you register the syntax of the subcommand so the players can see it

I didn't try this but I'm pretty sure it should work (with some caviats)
